### DISCUSSION TITLE
Contribution?

### DISCUSSION THREAD

**Discussion by similark** [estahn/k8s-image-swapper#101] [Q&A]
I was curious in adding some functionality but noticed there's no CONTRIBUTING.md in this repo. Also there's no makefile present so I have no idea even how this thing works??

**Comment by similark** [comment]
OK so there IS actually a CONTRIBUTING.md file but out of the box `make setup` (as described in that very same file) does not work as there's no makefile. What am I missing?

**Reply by estahn** [reply]
I have added a `Makefile` and adjusted the contributions guide. Running the wrapper is just a `go run ./main.go`.

However, I appreciate testing the webhook during development is a bit more complicated then outlined in the contributions guide. I will provide more details shortly, but a short version would be:

1. Create a kind cluster
2. Run [ngrok](https://ngrok.com/) for port 8443
3. ```
    helm repo add estahn https://estahn.github.io/charts/ 
    helm install estahn/k8s-image-swapper --set dev.enabled=true --set webhookURL=https://xxx.ngrok.io
    ```

This wires up the local running go program with the kind cluster.

Now start a pod via kubectl and you should see something in the logs.

**Reply by similark** [reply]
Thanks for the prompt answer, I appreciate it! I will give a try soon and get back with conclusions.

**Reply by estahn** [reply]
@similark I will try to provide a more elaborate guide with details today. Just out of curiosity, what would you like to contribute?

**Reply by similark** [reply]
Maybe adding harbor support but I'll have to make an estimate first ;)
